sql/planbuilder/ddl.go

```diff
@@ -1184,41 +1202,53 @@ func (b *Builder) modifySchemaTarget(inScope *scope, n sql.SchemaTarget, rt *pla
 }
 
 func (b *Builder) resolveSchemaDefaults(inScope *scope, schema sql.Schema) sql.Schema {
-	newSch := make(sql.Schema, len(schema))
-	for i, c := range schema {
-		col := c.Copy()
-		newSch[i] = col
-		if col.Default == nil || col.Default.Expr == nil {
-			continue
-		}
-		def, ok := col.Default.Expr.(*sql.UnresolvedColumnDefault)
-		if ok && def.String() == "" {
-			schema[i].Default = b.convertDefaultExpression(inScope, &ast.SQLVal{Val: []byte{}, Type: ast.StrVal}, schema[i].Type, schema[i].Nullable)
-			continue
-		} else if !ok {
-			continue
-		}
-		parsed, err := ast.Parse(fmt.Sprintf("SELECT %s", def))
-		if err != nil {
-			err := fmt.Errorf("%w: %s", sql.ErrInvalidColumnDefaultValue.New(def), err)
-			b.handleErr(err)
-		}
-		selectStmt, ok := parsed.(*ast.Select)
-		if !ok || len(selectStmt.SelectExprs) != 1 {
-			err := sql.ErrInvalidColumnDefaultValue.New(def)
-			b.handleErr(err)
-		}
-		expr := selectStmt.SelectExprs[0]
-		ae, ok := expr.(*ast.AliasedExpr)
-		if !ok {
-			err := sql.ErrInvalidColumnDefaultValue.New(def)
-			b.handleErr(err)
-		}
-		col.Default = b.convertDefaultExpression(inScope, ae.Expr, schema[i].Type, schema[i].Nullable)
+	newSch := schema.Copy()
+	for _, col := range newSch {
+		col.Default = b.resolveColumnDefaultExpression(inScope, col, col.Default)
+		col.Generated = b.resolveColumnDefaultExpression(inScope, col, col.Generated)
 	}
 	return newSch
 }
 
+func (b *Builder) resolveColumnDefaultExpression(inScope *scope, columnDef *sql.Column, colDefault *sql.ColumnDefaultValue) *sql.ColumnDefaultValue {
+	if colDefault == nil || colDefault.Expr == nil {
+		return colDefault
+	}
+
+	def, ok := colDefault.Expr.(*sql.UnresolvedColumnDefault)
+	if !ok {
+		// no resolution work to be done, return the original value
+		return colDefault
+	}
+
+	// Empty string is a special case, it means the default value is the empty string
+	// TODO: why isn't this serialized as ''
```

Comment:
i wonder if i made a mistake on the GMS implementation, if this is only a memory table problem

Reply:
It's possible, I'll add more tests.